File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/BalancedBrackets.py
class BalancedBrackets:
    def __init__(self, expr):
        self.stack = []
        self.left_brackets = "({["
        self.right_brackets = ")}]"
        self.expr = expr

    def clear_expr(self):
        self.expr = "".join(
            c for c in self.expr
            if c in self.left_brackets or c in self.right_brackets
        )

    def check_balanced_brackets(self):
        self.clear_expr()
        for brkt in self.expr:
            if brkt in self.left_brackets:
                self.stack.append(brkt)
            else:
                if not self.stack:
                    return False
                current_brkt = self.stack.pop()
                if current_brkt == '(' and brkt != ')':
                    return False
                if current_brkt == '{' and brkt != '}':
                    return False
                if current_brkt == '[' and brkt != ']':
                    return False
        return not self.stack

    def get_expr(self):
        return self.expr


if __name__ == "__main__":
    b = BalancedBrackets("a(b)c")
    print(b.check_balanced_brackets())

File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/BalancedBrackets.cpp
#include <stack>
#include <vector>
#include <string>

class BalancedBrackets {
private:
    std::string expr;
    std::stack<char> stack;
    std::vector<char> left_brackets;
    std::vector<char> right_brackets;

public:
    BalancedBrackets(const std::string& expr) : expr(expr) {
        left_brackets = {'(', '{', '['};
        right_brackets = {')', '}', ']'];
    }

    void clear_expr() {
        std::string new_expr;
        for (char c : expr) {
            if (std::find(left_brackets.begin(), left_brackets.end(), c) != left_brackets.end() ||
                std::find(right_brackets.begin(), right_brackets.end(), c) != right_brackets.end()) {
                new_expr += c;
            }
        }
        expr = new_expr;
    }

    bool check_balanced_brackets() {
        clear_expr();
        for (char Brkt : expr) {
            if (std::find(left_brackets.begin(), left_brackets.end(), Brkt) != left_brackets.end()) {
                stack.push(Brkt);
            } else {
                if (stack.empty()) {
                    return false;
                }
                char Current_Brkt = stack.top();
                stack.pop();
                if (Current_Brkt == '(' && Brkt != ')') return false;
                if (Current_Brkt == '{' && Brkt != '}') return false;
                if (Current_Brkt == '[' && Brkt != ']') return false;
            }
        }
        return stack.empty();
    }
};